### PR TITLE
Task detail view view model

### PR DESCRIPTION
**Pull request by punyabrata**
@nanisaladi Please review.

**Things taken care of in this pull request.**
_______________________________________________

1. Circular image-view changed back to regular image view on the _recycler list view_.
2. Style and logic changed for _original_ and _sale price_ values.
3. _You save text_ logic changed.
4. Expandable text-view introduced for the _short description_ text.
5. Overall styling changed for the _recycler list view_.
6. A detail view layout is created.
7. A method is created in retrofit to fetch data for a single item.
8. _DetailViewModel_ is created to facilitate the detail view.
9. Method added in the repository to fetch single item detail.
10. The detail view got a new UI too.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,6 +20,9 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".view.DetailActivity">
+
```
Comment (nanisaladi) [inline app/src/main/AndroidManifest.xml:24]: Let's a launchMode as singleTask.

File: app/src/main/java/com/learner/learndroid/view/DetailActivity.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        float productRating = Float.valueOf(item.getCustomerRating());
+        String productRatingValue = String.valueOf(productRating);
+        String productRatingText = getApplicationContext().getString(R.string.product_rating_text, productRatingValue);
+        ratingView.setRating(productRating);
+        ratingText.setText(productRatingText);
+
+        doneButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```
Comment (nanisaladi) [inline app/src/main/java/com/learner/learndroid/view/DetailActivity.java:134]: Suspect: Are we not using ButterKnife?

File: app/src/main/java/com/learner/learndroid/view/RecyclerViewAdapter.java [hunk 3]
```diff
@@ -72,37 +73,77 @@ public void onBindViewHolder(@NonNull RecyclerViewHolder recyclerViewHolder, fin
...
+    /**
+     * Sets the view holder on click listener.
+     *
+     * @param recyclerViewHolder View Holder.
+     * @param item               Item.
+     */
+    private void setViewHolderOnClickListener(@NonNull RecyclerViewHolder recyclerViewHolder, final Item item) {
+        recyclerViewHolder.parentLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```
Comment (nanisaladi) [inline app/src/main/java/com/learner/learndroid/view/RecyclerViewAdapter.java:89]: Shall we try this https://www.phunware.com/blog/android-clean-coding-approach/ ?

**Review by nanisaladi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/learner/learndroid/view/DetailActivity.java (hunk 2 above)
Comment (punyabrata) [reply app/src/main/java/com/learner/learndroid/view/DetailActivity.java:134]: That will be coming in the subsequent pull requests.

File: app/src/main/java/com/learner/learndroid/view/RecyclerViewAdapter.java (hunk 3 above)
Comment (punyabrata) [reply app/src/main/java/com/learner/learndroid/view/RecyclerViewAdapter.java:89]: We can. But I am thinking of doing this the ButterKnife way. So, I think let's try that and see how that goes. If still not good, we'll try this.

**Comment by punyabrata** [comment]
@nanisaladi Ready for review.

**Review by nanisaladi — APPROVED** [review]

**Pull request merged**